### ISSUE TITLE
tiny lookAt improvement

### ISSUE THREAD

**Issue by FyiurAmron** [JOML-CI/JOML#76]
in https://github.com/JOML-CI/JOML/blob/master/src/org/joml/Matrix4f.java#L5065 , you can reuse the already calculated values, so
```
        float dirX, dirY, dirZ;
        dirX = centerX - eyeX;
        dirY = centerY - eyeY;
        dirZ = centerZ - eyeZ;
        // Normalize direction
        float invDirLength = 1.0f / (float) Math.sqrt(
                  (centerX - eyeX) * (centerX - eyeX)
                + (centerY - eyeY) * (centerY - eyeY)
                + (centerZ - eyeZ) * (centerZ - eyeZ));
```
can become
```
        float dirX, dirY, dirZ;
        dirX = centerX - eyeX;
        dirY = centerY - eyeY;
        dirZ = centerZ - eyeZ;
        // Normalize direction
        float invDirLength = 1.0f / (float) Math.sqrt( dirX * dirX + dirY * dirY + dirZ * dirZ );
```

**Comment by FyiurAmron** [comment]
also, FWIF, you can improve the algo a tiny bit w.r.t. canonical OpenGL implementation - calculate -dir instead of dir, swapping the subtraction's order (i.e. calc eye-center instead of center-eye) - let's call it dir' - and then swap the order of the following crosses (take the advantage of the anticommutativity, i.e. the fact that a x b = - b x a), so that you get
right = dir x up = up x dir'
upn = right x dir = dir' x right

the trivial gain here is that, apart from simplifying the logic itself, you don't have to swap signs when assigning dir to output matrix later on, saving you 2 opcodes (actually, there are 3 saves, but 1 additional minus is needed in translation vector calc).

**Comment by httpdigest** [comment]
Awesome advice. Thanks! :)
Could you be so kind to provide a GitHub pull request with your changes? I could not completely follow you on your explanations. So let the code do the talking. :D

**Comment by httpdigest** [comment]
Okay, no PR necessary anymore. :)
I think this should do it. Is indeed easier to understand. Thanks!

**Comment by FyiurAmron** [comment]
I'm happy to be of use, mate. As a side note: the 'right' vector got to actually be a *left* vector now, since that change made it := -right ... yup, I used the name 'right' for it myself, since it has, obviously a right angle vs both direction and up, but now, after browsing OpenGL reference code I realized it was named "right" for real, as in "right wing", not as in "right approach".

I thus guess that renaming it & its components from 'right' to 'left' would probably make it easier for future generations to understand what's really going on there in that code.
